File: BookStoreApi/Database/Models/Book.cs

namespace BookStoreApi.Database.Models
{

    public class Book
    {
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string EnglishName { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public long Price { get; set; }
        public short PrintSeries { get; set; }
        public string ISBN { get; set; } = string.Empty;
        public string CoverType { get; set; } = string.Empty;
        public string Format { get; set; } = string.Empty;
        public string Pages { get; set; } = string.Empty;
        public string PublishYear { get; set; } = string.Empty;
        public string Publisher { get; set; } = string.Empty;
        public int AuthorId { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: BookStoreFront/src/app/models/book.ts
import { PaginationInfo } from './apiResponse';
import { Author } from './author';
import { Category } from './category';
import { Image } from './image';
import { Tag } from './tag';
import { Translator } from './translator';

export interface Book {
  id: number;
  name: string;
  englishName: string;
  description: string;
  price: number;
  printSeries: number; // سری چاپ
  isbn: string; // شابک
  coverType: string; // نوع جلد
  format: string; // قطع
  pages: string;
  publishYear: string;
  publisher: string;
  isRecommended: boolean;
  stock: number;
  authorId: number;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface BookAllData {
  id: number;
  name: string;
  englishName: string;
  description: string;
  price: number;
  printSeries: number; // سری چاپ
  isbn: string; // شابک
  coverType: string; // نوع جلد
  format: string; // قطع
  pages: string;
  publishYear: string;
  publisher: string;
  isRecommended: boolean;
  stock: number;
  quantity: number;
  authorId: number;
  createdAt?: Date;
  updatedAt?: Date;

  comments?: Comment[];
  translators?: Translator[];
  categories?: Category[];
  tags?: Tag[];
  images?: Image[];
  author?: Author;
}

export interface BookListResponse {
  books?: BookAllData[];
  pagination?: BPaginationInfo;
}

export interface BPaginationInfo extends PaginationInfo {}

export interface CreateBookRequest {
  name: string;
  englishName?: string;
  description?: string;
  price: number;
  printSeries: number;
  isbn: string;
  coverType: string;
  format: string;
  pages: number;
  publishYear: number;
  publisher: string;
  stock: number;
  authorId: number;
  images: Image[];
  translators?: number[];
  categories: number[];
  tags: number[];
}

export interface UpdateBookRequest {
  name: string;
  englishName?: string;
  description?: string;
  price: number;
  printSeries: number;
  isbn: string;
  coverType: string;
  format: string;
  pages: number;
  publishYear: number;
  publisher: string;
  stock: number;
  authorId: number;
  translators?: number[];
  categories: number[];
  tags: number[];
}
